src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,110 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final String INDEX_FOR_MESSAGE = "Index: ";
+    private static final String SIZE_FOR_MESSAGE = ", Size:";
+
+    private Object[] elementData;
+    private int size;
+
+    public ArrayList() {
+        this.elementData = new Object[DEFAULT_CAPACITY];
+        this.size = 0;
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == elementData.length || size > elementData.length) {
+            newSizeForArray(size);
+        }
+        elementData[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException(INDEX_FOR_MESSAGE
+                    + index + SIZE_FOR_MESSAGE + size);
+        }
+        if (size == elementData.length || size > elementData.length) {
+            newSizeForArray(size);
+        }
+        System.arraycopy(elementData,index, elementData,index + 1,size - index);
+        elementData[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        if (size + list.size() > elementData.length || size > elementData.length) {
+            newSizeForArray(size);
+        }
+        for (int i = 0; i < list.size();i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index < 0 || index >= size) {
+            throw new ArrayListIndexOutOfBoundsException(INDEX_FOR_MESSAGE
+                    + index + SIZE_FOR_MESSAGE + size);
+        }
+        return (T) elementData[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        if (index < 0 || index >= size) {
+            throw new ArrayListIndexOutOfBoundsException(INDEX_FOR_MESSAGE
+                    + index + SIZE_FOR_MESSAGE + size);
+        }
```

Comment:
Since you're reusing this code, you can put it in a separate method like in the example provided. Then, you can simply call the method instead of copying and pasting the same code.
```
private void isIndexLegal(int index) {
        if (!(index > -1 && index < size)) {
            throw new ArrayListIndexOutOfBoundsException(String.format(INDEX_ERR_MSG, index,size));
        }
    }
```